src/common/low_precision_transformations/src/gather.cpp

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "low_precision/gather.hpp"
+
+#include <memory>
+#include <ngraph/ngraph.hpp>
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+#include "low_precision/network_helper.hpp"
+#include "low_precision/rt_info/precision_preserved_attribute.hpp"
+#include "itt.hpp"
+
+namespace ngraph {
+namespace pass {
+namespace low_precision {
+
+std::shared_ptr<opset1::Constant> gatherDeqConstant(
+    const std::shared_ptr<ngraph::Node> gather,
+    const std::shared_ptr<ngraph::Node> dequantizationConstant) {
+    auto constant = ov::as_type_ptr<ngraph::opset1::Constant>(dequantizationConstant);
+    auto constantShape = constant->get_shape();
+    if (shape_size(constantShape) == 1ul) {
+        return NetworkHelper::toScalar(constant);
+    }
+
+    const auto rank = gather->get_input_partial_shape(0).size();
+    if (rank != constantShape.size()) {
+        // case when constShape without batch
+        while ((constantShape.size() > 1) && (constantShape.size() < rank)) {
+            constantShape.insert(constantShape.begin(), 1);
+        }
+        constant = ngraph::opset1::Constant::create(ngraph::element::i32, { constantShape.size() }, constantShape);
+    }
+
+    const int64_t axis = ov::as_type_ptr<opset1::Constant>(gather->get_input_node_shared_ptr(2))->cast_vector<int64_t>()[0];
+    const size_t normalizedAxis = normalize_axis(gather->get_friendly_name(), axis, gather->get_input_partial_shape(0).rank());
+
+    // Dequantization channel match with gather axis
+    if (constantShape[normalizedAxis] != 1ul) {
+        const auto gather7 = ov::as_type_ptr<ngraph::opset7::Gather>(gather);
+        if (gather7) {
+            const auto output = fold<ngraph::opset7::Gather>(
+                constant,
+                gather7->input_value(1),
+                gather7->input_value(2),
+                gather7->get_batch_dims());
+            constant = ov::as_type_ptr<opset1::Constant>(NetworkHelper::toScalarIfPossible(output));
+        }
+
+        const auto gather8 = ov::as_type_ptr<ngraph::opset8::Gather>(gather);
+        if (gather8) {
+            const auto output = fold<ngraph::opset8::Gather>(
+                constant,
+                gather8->input_value(1),
+                gather8->input_value(2),
+                gather8->get_batch_dims());
+            constant = ov::as_type_ptr<opset1::Constant>(NetworkHelper::toScalarIfPossible(output));
+        }
+    }
+    return constant;
+}
+
+GatherTransformation::GatherTransformation(const Params& params) : LayerTransformation(params) {
+    MATCHER_SCOPE(GatherTransformation);
+    auto gather = pattern::wrap_type<opset7::Gather, opset8::Gather>({ pattern::wrap_type<opset1::Multiply>(),
```

Comment:
Why the transformation doesn't support `opset1::Gather`?

Reply:
We can enable the transformation of converting Gather1 to Gather7 to use int8 precision, so it may not be very necessary to support Gather1?

Reply:
Enabled gather1 to gather7 transformation for GPU. Thanks!

Reply:
Curretnly GPU plugin is not fully supported for gather8. And also I remember that gather7 has some issues with non constant indices too, not very confident though. Could you let us know which tests were failing?

Reply:
I will enable opset1::Gather support for this transformation, so gather1to gather7 convert is not necessary for GPU now.